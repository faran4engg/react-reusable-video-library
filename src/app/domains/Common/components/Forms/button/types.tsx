import { ReactNode, MouseEvent } from 'react';
type ButtonType = 'button' | 'submit' | 'reset';

export interface ButtonProps {
  children: string;
  disabled?: boolean;
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
  type?: ButtonType;
  className?: string;
  icon?: ReactNode;
  withIcon?: boolean;
}
