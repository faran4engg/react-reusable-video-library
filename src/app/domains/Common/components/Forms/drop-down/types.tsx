type optionsType = {
  label?: string;
  value?: string;
  isDisabled?: boolean;
  isFixed?: boolean;
};

export interface DropDownProps {
  options: Array<optionsType>;
  onChange?: (e) => void;
  isMulti?: boolean;
  isClearable?: boolean;
  isSearchable?: boolean;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  value?: any;
}
