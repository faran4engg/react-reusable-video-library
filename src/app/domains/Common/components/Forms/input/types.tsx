import { ReactNode, ChangeEvent } from 'react';
type InputType = 'text' | 'email' | 'password' | 'number';
type AutoCompleteType = 'on' | 'off';

export interface InputProps {
  name?: string;
  id?: string;
  value?: string;
  list?: string;
  type?: InputType;
  onChange?: (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
  onKeyUp?: () => void;
  autocomplete?: AutoCompleteType;
  maxLength?: number;
  className?: string;
  placeholder?: string;
  readonly?: boolean;
  icon?: ReactNode;
  withIcon?: boolean;
}
