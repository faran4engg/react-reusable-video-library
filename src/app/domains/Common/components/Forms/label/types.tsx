import { ReactNode } from 'react';

export interface LabelProps {
  children: ReactNode;
  className?: string;
  id?: string;
  htmlFor: string;
}
