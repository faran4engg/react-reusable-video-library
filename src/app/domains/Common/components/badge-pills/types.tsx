export interface BadgePillsProps {
  name: string;
  onClick: () => void;
}
