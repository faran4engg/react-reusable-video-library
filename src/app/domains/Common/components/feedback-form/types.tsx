export interface FeedBackFormProps {
  setShowModalOnSuccess: (data: boolean) => void;
}
