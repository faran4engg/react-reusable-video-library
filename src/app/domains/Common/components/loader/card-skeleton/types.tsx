export interface CardSkeletonProps {
  noOfCards?: number;
}
