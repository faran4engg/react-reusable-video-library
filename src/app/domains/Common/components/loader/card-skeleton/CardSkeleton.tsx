import { FC } from 'react';
import SkeletonBody from './SkeletonBody';

import { CardSkeletonProps } from './types';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const CardSkeleton: FC<CardSkeletonProps> = ({ noOfCards }): any =>
  Array(noOfCards)
    .fill('-')
    .map((_, i) => (
      <div
        // eslint-disable-next-line react/no-array-index-key
        key={i}
        className="p-4 mx-auto w-full h-28 rounded-md shadow-md dark:bg-dark-mode bg-light-mode"
      >
        <SkeletonBody />
      </div>
    ));

CardSkeleton.defaultProps = {
  noOfCards: 1,
};

export default CardSkeleton;
