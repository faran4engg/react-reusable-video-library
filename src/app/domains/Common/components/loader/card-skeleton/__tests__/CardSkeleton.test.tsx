import { create } from 'react-test-renderer';

import CardSkeleton from '../CardSkeleton';
import { CardSkeletonProps } from './../types';

jest.mock('../SkeletonBody', () => 'SkeletonBody');

describe('CardSkeleton', () => {
  const renderComponent = (props: Partial<CardSkeletonProps> = {}) =>
    create(<CardSkeleton {...props} />);

  it('should render', () => {
    const component = renderComponent().toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render when noOfCards is provided', () => {
    const component = renderComponent({ noOfCards: 3 }).toJSON();
    expect(component).toMatchSnapshot();
  });
});
