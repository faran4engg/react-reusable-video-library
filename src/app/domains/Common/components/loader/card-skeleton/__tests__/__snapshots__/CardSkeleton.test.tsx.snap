// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSkeleton should render 1`] = `
<div
  className="p-4 mx-auto w-full h-28 rounded-md shadow-md dark:bg-dark-mode bg-light-mode"
>
  <SkeletonBody />
</div>
`;

exports[`CardSkeleton should render when noOfCards is provided 1`] = `
Array [
  <div
    className="p-4 mx-auto w-full h-28 rounded-md shadow-md dark:bg-dark-mode bg-light-mode"
  >
    <SkeletonBody />
  </div>,
  <div
    className="p-4 mx-auto w-full h-28 rounded-md shadow-md dark:bg-dark-mode bg-light-mode"
  >
    <SkeletonBody />
  </div>,
  <div
    className="p-4 mx-auto w-full h-28 rounded-md shadow-md dark:bg-dark-mode bg-light-mode"
  >
    <SkeletonBody />
  </div>,
]
`;
